sources/Makefile

```diff
@@ -1,3 +1,211 @@
-include Makefile.configurable
-include Makefile.users
-include Makefile.developers
+
+# ====================================
+# Variable Definitions & Configuration
+# ====================================
+
+NAME=alt-ergo
+
+# Include configuration makefile
+# Since there is a rule to create/update this makefile, make
+# should automatically run it and reload/restart if needed
+include Makefile.config
+
+# Some variables to help with adding
+# flags and/or renaming the dune binary
+DUNE=dune
+DUNE_FLAGS=--profile release
+
+# List the files:
+# - generated by rules in this makefile,
+# - used by the build process
+#
+# This excludes:
+# - .ml files generated by menhir or ocamllex
+#   (since they reside in dune specific directory)
+GENERATED=lib/util/config.ml
+
+
+# =======
+# Aliases
+# =======
+
+# IMPORTANT: this is the first rules, and as such the default
+# run when "make is called", keep it as the first rule
+world: all
+
+# Small alias to re-generate necessary files for the build
+gen: $(GENERATED)
+
+# Convenient alias for running all configuration steps
+conf: Makefile Makefile.config lib/util/config.ml
+
+# Alias for generated artifacts
+clean: dune-clean
+
+# Alias to remove all generated files
+distclean: makefile-distclean release-distclean
+
+# declare these aliases as phony
+.PHONY: world gen conf clean distclean
+
+
+
+# ===========
+# Build rules
+# ===========
+
+# Build the alt-ergo lib
+lib: gen
+	$(DUNE) build $(DUNE_FLAGS) lib/AltErgoLib.{cma,cmxa,cmxs}
+
+# Build the cli/text alt-ergo bin
+bin: gen
```

Comment:
Can we have/need a link for binaries in sources/ ?

Reply:
@OCamlPro-Iguernlala suggested having symbolic links in the sources directory that would point to the relevant locations

Reply:
I create the symbolics links (though they may point to non-existant files if the bins have not been compiled yet).